### PR TITLE
feat: support aws sdk v 2 java tracer (Rd 12691)

### PR DESCRIPTION
**Pull request by nadav3396**

### CODE REVIEW

File: src/main/java/io/lumigo/core/utils/AwsSdkV2Utils.java [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import lombok.experimental.UtilityClass;
+import software.amazon.awssdk.services.dynamodb.model.AttributeValue;
+
+@UtilityClass
+public class AwsSdkV2Utils {
```
Comment (nadav3396) [inline src/main/java/io/lumigo/core/utils/AwsSdkV2Utils.java:11]: Add tests to this class

File: src/main/java/io/lumigo/core/SpansContainer.java [hunk 2]
```diff
@@ -354,14 +360,88 @@ public void addHttpSpan(Long startTime, Request<?> request, Response<?> response
...
+                                                .statusCode(context.httpResponse().statusCode())
+                                                .build())
+                                .build());
+
+        Logger.debug(
+                "Trying to extract aws custom properties for service: "
+                        + executionAttributes.getAttribute(SdkExecutionAttribute.SERVICE_NAME));
+        AwsSdkV2ParserFactory.getParser(
+                        executionAttributes.getAttribute(SdkExecutionAttribute.SERVICE_NAME))
+                .safeParse(httpSpan, context);
```
Comment (moshe-shaham-lumigo) [inline src/main/java/io/lumigo/core/SpansContainer.java:432]: are you sure nothing can be shared with v1 functions? painful to see this code duplication...

**Review by moshe-shaham-lumigo — APPROVED** [review]
LGTM

File: src/main/java/io/lumigo/core/SpansContainer.java (hunk 2 above)
Comment (nadav3396) [reply src/main/java/io/lumigo/core/SpansContainer.java:432]: not with the time limits we have now

**Pull request merged**